superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndMetricSelect.tsx

```diff
@@ -96,16 +96,17 @@ const getMetricsMatchingCurrentDataset = (
     return values;
   }
   return values.reduce((acc: ValueType[], metric) => {
-    if (
-      (typeof metric === 'string' || (metric as Metric).metric_name) &&
-      (areSavedMetricsEqual ||
+    if (typeof metric === 'string' || (metric as Metric).metric_name) {
```

Comment:
Consideration: now that we have those nifty new type guards, would we be able to introduce those in this PR?

Also, could we add the return type on line 90 (I'm guessing it's now `ValueType[]`)?

Reply:
I'd love to do that, but it seems that we miss `isAdhocMetric` type guard. Also, our `isSavedMetric` checks if value is a string, so it won't cover the `(metric as Metric).metric_name` case. Maybe we can circle back to it once monorepo is merged and we can easily add new type guards?

Reply:
yes, absolutely, let's do that 👍